File: login/static/login/javascript/checkValidity.js
//checkValidity.js
//입력값의 형식 유효성 체크

function isEmptyInput(data){ //입력값이 공백인지 확인
    if (data === '')
        return true;
    return false;
}

function isValidEmail(email){ //이메일 형식 유효성 체크
    var emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/; //이메일 형식 regex
    return emailRegex.test(email);
}

function isValidPassword(password){ //비밀번호 형식 유효성 체크
    //if (password === '')
    //    return false;     //일단 공백인 경우에만 false 반환, 추후 비밀번호 규칙이 정해지는 경우 수정
    return true;
}


//입력값이 유효한지 확인
//반환 형식 : [(유효성 여부), (메시지)]
//유효성 여부 | 0 : 이상 없음, 1 : 공백 입력, 2 : 기타

function userIdCheck(id){ //아이디 유효성 체크
    if (isEmptyInput(id))
        return [1, "이메일을 입력하세요"];
    if (!isValidEmail(id))
        return [2, "이메일 형식이 올바르지 않습니다"];
    return [0];
}

function passwordCheck(password){ //비밀번호 유효성 체크
    if (isEmptyInput(password))
        return [1, "비밀번호를 입력하세요"];
    if (!isValidPassword(password))
        return [2, "비밀번호 형식이 올바르지 않습니다"];
    return [0];
}

function passwordConfirmCheck(pwConfirm, pw){ //비밀번호 확인 유효성 체크
    if (isEmptyInput(pwConfirm))
        return [1, "비밀번호를 재입력하세요"];
    if (pwConfirm !== pw)
        return [2, "비밀번호가 일치하지 않습니다"];
    return [0];
}

function verificationCheck(verification){ //인증 코드 유효성 체크
    if (isEmptyInput(verification))
        return [1, "인증 코드를 입력하세요"];
    return [0];
}

function userNameCheck(userName){ //유저 네임 유효성 체크
    if (isEmptyInput(userName))
        return [1, "사용자 이름을 입력하세요"];
    return [0];
}

File: login/utils/checkValidity.py
from login.models import User
from . import pwHash, signUpManager
import re

def isEmptyInput(data): #입력값이 공백인지 확인
    if data == "":
        return True
    return False

def isValidEmail(email): #이메일 형식 유효성 체크
    emailRegex = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$' #이메일 형식 체크를 위한 regex
    if not re.search(emailRegex, email):
        return False
    return True

def isUserExist(id): #DB에 해당 id(email)을 가진 사용자가 존재하는지 확인
    return User.objects.filter(email = id).exists()

def isValidPassword(pw): #비밀번호 형식 유효성 체크
    return True

#입력된 데이터 유효성 체크
#반환형식 : (validity, message)
#validity : False(유효하지 않음), True(유효함)

def userIdCheck(id, userShouldBeExist):
    if id == "none":
        return (False, "none")
    
    if isEmptyInput(id):
        return (False, "이메일을 입력하세요")
    if not isValidEmail(id):
        return (False, "이메일 형식이 올바르지 않습니다")
    if isUserExist(id) and not userShouldBeExist:
        return (False, "중복되는 이메일이 이미 존재합니다")
    elif not isUserExist(id) and userShouldBeExist:
        return (False, "사용자가 존재하지 않습니다")
    '''
    if userShouldBeExist and not isUserExist(id): #(true, false)
        return "사용자가 존재하지 않습니다"
    if not userShouldBeExist and isUserExist(id): #(false, true)
        return "중복되는 이메일이 이미 존재합니다"
    '''
    return (True, "none")

def passwordCheck(pw, userPw=None): #pw : 사용자가 입력한 패스워드, userPw : DB의 User 테이블에 저장된 패스워드
    if pw == "none":
        return (False, "none")
    
    if isEmptyInput(pw):
        return (False, "비밀번호를 입력하세요")
    if not isValidPassword(pw):
        return (False, "비밀번호 형식이 올바르지 않습니다")
    if userPw != None and not pwHash.comparePW(pw, userPw):
        return (False, "비밀번호가 일치하지 않습니다")
    
    return (True, "none")

def passwordConfirmCheck(pwConf, pw):
    if pwConf == "none":
        return (False, "none")
    
    if isEmptyInput(pwConf):
        return (False, "비밀번호를 재입력하세요")
    if pwConf != pw:
        return (False, "비밀번호가 일치하지 않습니다")

    return (True, "none")

def verificationCheck(code):
    if code == "none":
        return (False, "none")
    
    if isEmptyInput(code):
        return (False, "인증 코드를 입력하세요")
    if not signUpManager.isCorrectVerificationCode(code):
        return (False, "코드가 일치하지 않습니다")
    
    return (True, "none")

def userNameCheck(name):
    if name == "none":
        return (False, "none")
    
    if isEmptyInput(name):
        return (False, "사용자 이름을 입력하세요")
    
    return (True, "none")